Fix failing tests and bad program behavior when missing files

StephenMcConnel opened the pull request:
This is a follow-on fix for the fix of BL-5050.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/1868)
<!-- Reviewable:end -->

hatton commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/BloomExe/Book/Book.cs, line 674 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868#-KuzeXRQzAu3NYcD_wJ-:-KuzeXRQzAu3NYcD_wJ0:bbsznvw) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/a80aedc3061b99df07f4c0f19f61552d7f367048/src/BloomExe/Book/Book.cs#L674)):*
> ```Smalltalk
> 		public virtual string FolderPath
> 		{
> 			get { return _storage?.FolderPath; }
> ```

when is _storage null, and why? A quick look at users of FolderPath suggest much pain if it starts being null.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/BloomExe/web/ServerBase.cs, line 461 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868#-Kuzf_4QJSeesVa-l9h_:-Kuzf_4QJSeesVa-l9ha:b-1o2go4) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/a80aedc3061b99df07f4c0f19f61552d7f367048/src/BloomExe/web/ServerBase.cs#L461)):*
> ```Smalltalk
> 				Logger.WriteEvent("**{0}: File Missing: {1}", GetType().Name, localPath);
> 			}
> 			info.WriteError(404);
> ```

I guess this does the same thing, i.e. avoids Logging the event. I guess the WriteError tells the browser not to expect a picture, so it won't tell us it's missing? I suppose if I was set on separating ShouldReport... from Report Missing File, we'd have to have a multi-if situation:
if (!ProcessRequest(info)
{
    if (ShouldReport...(info))
    {
        ReportMissingFile(info)
    }
    info.WriteError(404);
}
Otherwise ReportMissingFile() might not...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868)*
<!-- Sent from Reviewable.io -->

StephenMcConnel commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/BloomExe/Book/Book.cs, line 674 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868#-KuzeXRQzAu3NYcD_wJ-:-KuzidV7ZLNd8wBaVoVH:beguq5p) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/a80aedc3061b99df07f4c0f19f61552d7f367048/src/BloomExe/Book/Book.cs#L674)):*
<details><summary><i>Previously, hatton (John Hatton) wrote…</i></summary><blockquote>

when is _storage null, and why? A quick look at users of FolderPath suggest much pain if it starts being null.
</blockquote></details>

When moq'd for tests _storage never gets set.  And until now, we never felt any pain in the tests...
I added a comment about tests not setting _storage, and that this change suffices to pass the tests.

---

*[src/BloomExe/web/ServerBase.cs, line 461 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868#-Kuzf_4QJSeesVa-l9h_:-KuziphDvqCdrtMraTTw:b-ez81y2) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/a80aedc3061b99df07f4c0f19f61552d7f367048/src/BloomExe/web/ServerBase.cs#L461)):*
<details><summary><i>Previously, gmartin7 (Gordon Martin) wrote…</i></summary><blockquote>

I guess this does the same thing, i.e. avoids Logging the event. I guess the WriteError tells the browser not to expect a picture, so it won't tell us it's missing? I suppose if I was set on separating ShouldReport... from Report Missing File, we'd have to have a multi-if situation:
if (!ProcessRequest(info)
{
    if (ShouldReport...(info))
    {
        ReportMissingFile(info)
    }
    info.WriteError(404);
}
Otherwise ReportMissingFile() might not...
</blockquote></details>

That is an alternative structure to accomplish what is needed.  I've changed the code to this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1868#-:-KuzqrkkqxM-0ZeKDnhq:bnfp4nl)*
<!-- Sent from Reviewable.io -->
